controllers/PlagiarismEulaAcceptanceHandler.inc.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file plugins/generic/plagiarism/controllers/PlagiarismEulaAcceptanceHandler.inc.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class PlagiarismEulaAcceptanceHandler
+ * @ingroup plugins_generic_plagiarism
+ *
+ * @brief Handle the reconfirmation and acceptance of iThenticate EULA
+ */
+
+import("plugins.generic.plagiarism.controllers.PlagiarismComponentHandler");
+import("plugins.generic.plagiarism.IThenticate");
+
+class PlagiarismEulaAcceptanceHandler extends PlagiarismComponentHandler {
+
+	/**
+	 * Handle the presentation of iThenticate EULA right before the submission final stage
+	 *
+	 * @param array $args
+	 * @param Request $request
+	 * 
+	 * @return void
+	 */
+	public function handle($args, $request) {
+		$request = Application::get()->getRequest();
+		$context = $request->getContext();
+		$user = $request->getUser();
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /** @var SubmissionDAO $submissionDao */
+		$submission = $submissionDao->getById($args['submissionId']);
```

Comment:
Be careful with calls like this! End users can specify an arbitrary submission ID, including something that exists in a totally different journal.

Reply:
@asmecher I have added few auth policies to handle it . please check if that is the correct approach as I am still not very knowledgable about the authorization policy toolset . However still need checking for the OMP and OPS .

Reply:
done making OMP and OPS compatible .